File: 1-two-sum/1-two-sum.java
import java.util.HashSet; 
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> arr = new HashMap<>();
        for(int i=0; i<nums.length; i++){
            int sub = target - nums[i];
            if(arr.containsKey(sub)){
                return new int[] {arr.get(sub), i};
            }
            arr.put(nums[i], i);
        }
        return null;
    }
}

File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        result = []
        seen = {}
        
        for i in range(len(nums)):
            remaining = target - nums[i]
            
            if remaining in seen: 
                return [seen[remaining], i]
            else:
                seen[nums[i]] = i
    